discovery/seata-discovery-sofa/src/main/java/io/seata/discovery/registry/sofa/SofaRegistryServiceImpl.java

```diff
@@ -0,0 +1,335 @@
+    }
+
+    @Override
+    public void unsubscribe(String cluster, SubscriberDataObserver listener) throws Exception {
+        getRegistryInstance().unregister(cluster, registryProps.getProperty(PRO_GROUP_KEY), RegistryType.SUBSCRIBER);
+    }
+
+    @Override
+    public List<InetSocketAddress> lookup(String key) throws Exception {
+        Configuration config = ConfigurationFactory.getInstance();
+        String clusterName = config.getConfig(PREFIX_SERVICE_ROOT + CONFIG_SPLIT_CHAR + PREFIX_SERVICE_MAPPING + key);
+        if (null == clusterName) {
+            return null;
+        }
+        if (!LISTENER_SERVICE_MAP.containsKey(clusterName)) {
+            subscribe(clusterName, new SubscriberDataObserver() {
+                @Override
+                public void handleData(String dataId, UserData data) {
+                    Map<String, List<String>> instances = data.getZoneData();
+                    if (null == instances && null != CLUSTER_ADDRESS_MAP.get(clusterName)) {
+                        CLUSTER_ADDRESS_MAP.remove(clusterName);
+                    } else {
+                        List<InetSocketAddress> tranformData = flatData(instances);
+                        List<InetSocketAddress> newAddressList = new ArrayList<>();
+                        newAddressList.addAll(tranformData);
+                        CLUSTER_ADDRESS_MAP.put(clusterName, newAddressList);
+                    }
+                    respondRegistries.countDown();
+                }
+            });
+
+            //wait max for first lookup
+            final String property = registryProps.getProperty(PRO_ADDRESS_WAIT_TIME_KEY);
+            respondRegistries.await(Integer.parseInt(property), TimeUnit.MILLISECONDS);
+
+        }
+        return CLUSTER_ADDRESS_MAP.get(clusterName);
+    }
+
+
+    private List<InetSocketAddress> flatData(Map<String, List<String>> instances) {
+        List<InetSocketAddress> result = new ArrayList<InetSocketAddress>();
+
+        for (Map.Entry<String, List<String>> entry : instances.entrySet()) {
+            for (String str : entry.getValue()) {
+                String ip = StringUtils.substringBefore(str, HOST_SEPERATOR);
+                String port = StringUtils.substringAfter(str, HOST_SEPERATOR);
+                InetSocketAddress inetSocketAddress = new InetSocketAddress(ip, Integer.parseInt(port));
+                result.add(inetSocketAddress);
+            }
+        }
+        return result;
+    }
+
+    @Override
+    public void close() throws Exception {
+    }
+
+    private void validAddress(InetSocketAddress address) {
+        if (null == address.getHostName() || 0 == address.getPort()) {
+            throw new IllegalArgumentException("invalid address:" + address);
+        }
+    }
+
+    private static Properties getNamingProperties() {
+        Properties properties = new Properties();
+        if (null != System.getProperty(PRO_SERVER_ADDR_KEY)) {
```

Comment:
I think Missing prefix like `-Dregistry.serverAddr`  but not `-DserverAddr`

Reply:
ok